test_aodndata/soop/test_soop_ba.py

```diff
@@ -0,0 +1,67 @@
+from aodncore.testlib import HandlerTestCase
+from aodndata.soop.soop_ba import SoopBaHandler
+
+TEST_ROOT = os.path.join(os.path.dirname(__file__))
+GOOD_NC = os.path.join(TEST_ROOT,
+                       'IMOS_SOOP-BA_A_20160116T152139Z_VKAD_FV02_Antarctic-Discovery-ES60-38_END-20160129T062056Z_C-20170227T055245Z.nc')
+GOOD_ZIP = os.path.join(TEST_ROOT, 'Antarctic-Discovery_20160116-20160129.zip')
+
+
+class TestSoopBaHandler(HandlerTestCase):
+    """It is recommended to inherit from the HandlerTestCase class (which is itself a subclass of the standard
+       unittest.TestCase class). This provides some useful methods and properties to shortcut some common test
+       scenarios.
+    """
+
+    # This is a "boilerplate" method that must appear in each test case
+    #  in order to correctly inherit from the HandlerTestCase class
+    def setUp(self):
+        # set the handler_class attribute to your handler (as imported above)
+        self.handler_class = SoopBaHandler
+        super(TestSoopBaHandler, self).setUp()
+
+    def test_good_ba_file(self):
+        # we expect this to succeed, so if the handler experiences an error, it is considered a
+        # "failed test"
+        handler = self.run_handler(GOOD_NC)
+
+        f = handler.file_collection[0]
+        # self.assertEqual(f.check_type, PipelineFileCheckType.NC_COMPLIANCE_CHECK)
+        self.assertEqual(f.publish_type, PipelineFilePublishType.HARVEST_UPLOAD)
+        self.assertEqual(f.name, os.path.basename(GOOD_NC))
+        self.assertEqual(f.dest_path,
+                         'IMOS/SOOP/SOOP-BA/VKAD_Antarctic-Discovery/Antarctic-Discovery_20160116-20160129/'
+                         + f.name)
+        self.assertTrue(f.is_stored)
+
+    def test_good_nc_zip(self):
+
+        handler = self.handler_class(GOOD_ZIP)
+
+        handler.run()
+
+        nc_files = handler.file_collection.filter_by_attribute_value('extension', '.nc')
+
+        for f in nc_files:
+            # self.assertEqual(f.check_type, PipelineFileCheckType.NC_COMPLIANCE_CHECK)
+            self.assertEqual(f.publish_type, PipelineFilePublishType.HARVEST_UPLOAD)
+            self.assertEqual(f.name, os.path.basename(GOOD_NC))
+            self.assertEqual(f.dest_path,
+                             'IMOS/SOOP/SOOP-BA/VKAD_Antarctic-Discovery/Antarctic-Discovery_20160116-20160129/' + f.name
+                             )
+
+        non_nc = handler.file_collection.filter_by_attribute_regex('extension', '[.](?!nc$)$')
+        print non_nc
+        for f in non_nc:
+            print 'non nc %s' % f.name
+            self.assertEqual(f.dest_path,
+                             'IMOS/SOOP/SOOP-BA/VKAD_Antarctic-Discovery/Antarctic-Discovery_20160116-20160129/' + f.name)
+            self.assertEqual(f.check_type, PipelineFileCheckType.NONEMPTY_CHECK)
+            self.assertEqual(f.publish_type, PipelineFilePublishType.UPLOAD_ONLY)
+
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
Also you need to add your handler here
https://github.com/aodn/python-aodndata/blob/master/setup.py#L21